### ISSUE TITLE
Add other notifcation agents besides email (like slack/rocketchat/mattermost)

### ISSUE THREAD

**Issue by vulcanjedi** [monicahq/monica#2303]
(Note: you don't need to follow this template, nor to keep headlines or bold sentences - they are just there to guide you. Feel free to delete everything. We review every issue even if we don't immediately respond.)

Thanks for filing an issue and for your interest in the project.

**Is your feature request related to a problem? Please describe.**
A clear and concise description of what the problem is. Ex.
* I'm always frustrated when [...]
* Current feature X is awesome but doesn't fill the need of [...]
* Monica doesn't have feature X which is essential as [...]

**Describe the solution you'd like**
Email notifcations are great but i think if I got a push notification on mobile it woudl be nice.  Such as a Slack notification vs Email

**Describe alternatives you've considered (optional)**
Alternatives would include Rocketchat, Zulip, Mattermost, ...etc

**Additional context**
These are just some of the typical agents I see in other webapps

**Comment by Underestimated** [comment]
Yeah just flat webhooks would be useful, as they can be integrated into IFTTT too

**Comment by grahamPegNetwork** [comment]
+1 for this. I'm currently evaluating Monica, but the biggest thing that stands out is that I don't want email notifications. I'd rather them somewhere else I can manage. Send them via webhook to Zapier/n8n/IFTTT/custom node scripts etc. 

Notifications about contact birthdays is probably one of my biggest use cases, but email? Now let me toss them through webhooks to Trello or some other sort of task manager and we're talking. I could add in a pushbullet notification while we're at it, now that's much more easy to digest.

**Comment by virtadpt** [comment]
It'd be really nice to be able to hit a REST API or send an XMPP message.

**Comment by RCheesley** [comment]
Another +1 for some way to integrate with other platforms such as Zapier or IFTTT as a starting point to get reminders/tasks out into other tools :)

**Comment by immanuelfodor** [comment]
I'd be happy with plain webhook support at first and then I make something out of the payload myself. On the other hand, Slack payloads seem to be rather well supported across different tools.

**Comment by immanuelfodor** [comment]
Maybe this comes handy: https://laravel-notification-channels.com/webhook/

Also, I've found similar issues: https://github.com/monicahq/monica/issues/68 https://github.com/monicahq/monica/issues/318

**Comment by dichvuhuuich** [comment]
> I'm using n8n which does have a module for checking an IMAP mailbox, so technically I can make it work, but it's a bit more complicated than it should be. I'll probably use it in the short term.

I'm using n8n, too, hope monica will have webhook to create trigger node in n8n. I've create feature request here, please go and vote to create Monica node in n8n https://community.n8n.io/t/monica-personal-crm-intergration/1566

**Comment by nhymxu** [comment]
Maybe too much code change to support webhook to widely support many application.
But I found that laravel support custom channel (email is default channel of laravel)
https://laravel.com/docs/8.x/notifications#custom-channels

So I think I can create new telegram channel, user just enter telegram bot id to `.env` file (same as setup email) to get notification.

First version is same content as email.

What do you think 🤔
